src/VoltageCalibration.cc

```diff
@@ -1058,64 +1053,51 @@ double * mattak::applyVoltageCalibration (int N, const int16_t * in, double * ou
     return 0;
   }
 
-  bool is2ndHalfWindows;
-  int nwindows = N / mattak::k::radiant_window_size;
-  int isamp;
+  int nSamplesPerGroup = mattak::k::num_radiant_samples;
+  int nWindowsPerGroup = mattak::k::radiant_windows_per_buffer;
   int i = 0;
-  int j;
+  int isamp, isamp_A, isamp_B;
 
-  if (start_window > 15)
+  if (isOldFirmware)
   {
-    is2ndHalfWindows = true;
-    j = start_window - mattak::k::radiant_windows_per_buffer;
+    nSamplesPerGroup /= 2;
+    nWindowsPerGroup /= 2;
+    isamp_A = (start_window / nWindowsPerGroup) * nSamplesPerGroup;
   }
   else
   {
-    is2ndHalfWindows = false;
-    j = start_window;
+    isamp_A = (start_window >= nWindowsPerGroup) * nSamplesPerGroup;
   }
 
-
-  for (int iwindow = 0; iwindow < nwindows; iwindow++)
+  for (int i = 0; i < N; i++)
   {
-    if (isOldFirmware)
-    {
-      // Old Firmware
-      // Wrap around if j % 16 = 0
-      if (j % mattak::k::radiant_windows_per_buffer == 0) j = 0;
-      if (inRange(0,4,j) || inRange(8,12,j)) isamp = (j+3) * mattak::k::radiant_window_size;
-      if (inRange(5,7,j) || inRange(13,15,j)) isamp = (j-5) * mattak::k::radiant_window_size;
-    }
-    else
-    {
-      // New Firmware
-      // Wrap around if j % 16 = 0
-      if (j % mattak::k::radiant_windows_per_buffer == 0) j = 0;
-      if (inRange(0,12,j)) isamp = (j+3) * mattak::k::radiant_window_size;
-      if (inRange(13,15,j)) isamp = (j-13) * mattak::k::radiant_window_size;
-    }
+#ifndef MATTAK_VECTORIZE
 
-    if (is2ndHalfWindows) isamp += mattak::k::num_radiant_samples;
+    isamp_B = (i + start_window * mattak::k::radiant_window_size) % nSamplesPerGroup;
 
-    j++;
+    isamp = isamp_A + isamp_B;
 
-#ifndef MATTAK_VECTORIZE
-    for (int k = 0; k < mattak::k::radiant_window_size; k++)
+    const double *params = packed_fit_params + isamp * (fit_order+1);
+
+    double *residTablePerSamp_adc;
+    if (isUsingResid)
     {
-      const double *params = packed_fit_params + isamp * (fit_order+1);
+      residTablePerSamp_adc = adcTablePerSample(fit_order, nResidPoints, params, packed_aveResid_volt, packed_aveResid_adc);
+    }
 
-      double *residTablePerSamp_adc;
-      if (isUsingResid) residTablePerSamp_adc = adcTablePerSample(fit_order, nResidPoints, params, packed_aveResid_volt, packed_aveResid_adc);
+    double adc = in[i];
+    if (isUsingResid)
+    {
```

Comment:
I personally would put the `{` after the if condition (but this is personal preference.

Reply:
Yeah I've already put those brackets since you suggested me to do so yesterday

Reply:
my code generally follows Allman indentation style, FWIW

Reply:
I also think Allman style looks better. Regarding to adding brackets, I've seen that sometimes when there is only one short line after the if condition, people just don't add brackets since it's just one short line, that's why I did it.

Reply:
Okay than we go with Allman style fine! Whether to use brackets or not for a one line command in an if condition is again a matter of style and we should just be consistent. However only if the command is in the next line, I am strongly opposing having in one line all together.